service/service.go

```diff
@@ -22,27 +22,31 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/multierr"
+	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service/internal"
 	"go.opentelemetry.io/collector/service/internal/extensions"
 	"go.opentelemetry.io/collector/service/internal/pipelines"
+	"go.opentelemetry.io/collector/service/internal/telemetry"
 	"go.opentelemetry.io/collector/service/internal/telemetrylogs"
 )
 
 // service represents the implementation of a component.Host.
 type service struct {
-	buildInfo component.BuildInfo
-	config    *Config
-	telemetry component.TelemetrySettings
-	host      *serviceHost
+	buildInfo    component.BuildInfo
+	config       *Config
+	telemetry    component.TelemetrySettings
+	host         *serviceHost
+	colTelemetry *colTelemetry
```

Comment:
It is a bit confusing that we have `telemetry` and `colTelemetry` fields. Later in the code when they are used it is not clear what the difference is. 

Some renaming suggestions:
- `service.colTelemetry` -> `service.telemetryInitializer`
- `service.telemetry` -> `service.telemetrySettings` or (`TelemetrySettings` -> `TelemetryProviders` and `service.telemetry` -> `service.telemetryProviders`)

Reply:
Done, longer names (Java ftw 👍🏽  )